File: app/src/main/java/org/metabrainz/mobile/data/sources/api/entities/mbentity/Event.java
package org.metabrainz.mobile.data.sources.api.entities.mbentity;

import androidx.annotation.NonNull;

import com.google.gson.annotations.SerializedName;

import org.metabrainz.mobile.data.sources.api.entities.LifeSpan;

public class Event extends MBEntity {

    //TODO: Add Relations Field

    private String name;
    @SerializedName("life-span")
    private LifeSpan lifeSpan;
    private String type;
    private String time;

    @NonNull
    @Override
    public String toString() {
        return "Event{" +
                "name='" + name + '\'' +
                ", disambiguation='" + disambiguation + '\'' +
                ", lifeSpan=" + lifeSpan +
                ", type='" + type + '\'' +
                ", time='" + time + '\'' +
                '}';
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public LifeSpan getLifeSpan() {
        return lifeSpan;
    }

    public void setLifeSpan(LifeSpan lifeSpan) {
        this.lifeSpan = lifeSpan;
    }

    public String getType() {
        return type;
    }

    public void setType(String type) {
        this.type = type;
    }

    public String getTime() {
        return time;
    }

    public void setTime(String time) {
        this.time = time;
    }
}

File: app/src/main/java/org/metabrainz/android/model/mbentity/Event.kt
package org.metabrainz.android.model.mbentity

import com.google.gson.annotations.SerializedName
import org.metabrainz.android.data.sources.api.entities.LifeSpan

class Event : org.metabrainz.android.model.mbentity.MBEntity() {
    //TODO: Add Relations Field
    var name: String? = null

    @SerializedName("life-span")
    var lifeSpan: LifeSpan? = null
    var type: String? = null
    var time: String? = null
    override fun toString(): String {
        return "Event{" +
                "name='" + name + '\'' +
                ", disambiguation='" + disambiguation + '\'' +
                ", lifeSpan=" + lifeSpan +
                ", type='" + type + '\'' +
                ", time='" + time + '\'' +
                '}'
    }
}